lib/active_merchant/billing/gateways/payu_latam.rb

```diff
@@ -0,0 +1,393 @@
+
+        case action
+        when 'auth'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'VOID')
+          add_reference(post, authorization)
+        when 'purchase', 'capture'
+          add_credentials(post, 'SUBMIT_TRANSACTION')
+          add_transaction_type(post, 'REFUND')
+          add_reference(post, authorization)
+        else
+          message = "Unsupported operation: successful Purchase, Authorize and unsettled Capture transactions can only be voided."
+          return Response.new(false, message)
+        end
+
+        commit('void', post)
+      end
+
+      # Gateway supports authorize for certain countries only,
+      # hence a combination of purchase and void for verify
+      #
+      # Each country has different minimum amounts,
+      # hence 100.00 instead of typical 1.00
+      def verify(credit_card, options={})
+        MultiResponse.run(:use_first_response) do |r|
+          r.process { purchase(10000, credit_card, options) }
+          r.process(:ignore_result) { void(r.authorization, options) }
+        end
+      end
+
+      def store(payment_method, options = {})
+        post = {}
+
+        add_credentials(post, 'CREATE_TOKEN')
+        add_payment_method_to_be_tokenized(post, payment_method)
+
+        commit('store', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript.
+          gsub(%r((\"creditCard\\\":{\\\"number\\\":\\\")\d+), '\1[FILTERED]').
+          gsub(%r((\"securityCode\\\":\\\")\d+), '\1[FILTERED]').
+          gsub(%r((\"apiKey\\\":\\\")\w+), '\1[FILTERED]')
+      end
+
+      private
+
+      def add_credentials(post, command)
+        post[:test] = test? unless command == 'CREATE_TOKEN'
+        post[:language] = 'en'
+        post[:command] = command
+        merchant = {}
+        merchant[:apiLogin] = @options[:api_login]
+        merchant[:apiKey] = @options[:api_key]
+        post[:merchant] = merchant
+      end
+
+      def add_transaction_type(post, type)
+        transaction = {}
+        transaction[:type] = type
+        post[:transaction] = transaction
+      end
+
+      def add_order(post, options)
+        order = {}
+        order[:accountId] = @options[:account_id]
+        order[:referenceCode] = options[:order_id] || SecureRandom.hex(16)
+        order[:description] = options[:description] || 'unspecified'
+        order[:language] = 'en'
+        post[:transaction][:order] = order
+      end
+
+      def add_buyer(post, options)
+        buyer = {}
+        if dni_number = options[:buyer_dni_number] || options[:dni_number]
+          buyer[:dniNumber] = dni_number
+        end
+        buyer[:cnpj] = options[:cnpj] if options[:cnpj]
+        buyer[:emailAddress] = options[:shipping_email] || options[:email] || 'unspecified@example.com'
+        if address = options[:shipping_address] || options[:billing_address] || options[:address]
```

Comment:
Thinking we can just rely on `options[:shipping_address]` here and disregard the other 2.